modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteServerImpl.java

```diff
@@ -119,6 +119,14 @@ public class IgniteServerImpl implements IgniteServer {
     @Nullable
     private CompletableFuture<Void> restartOrShutdownFuture;
 
+    /**
+     * Future of the last internal restart future ({@code null} if no internal restarts where made).
+     *
+     * <p>Guarded by {@link #restartOrShutdownMutex}.
+     */
+    @Nullable
+    private CompletableFuture<Void> restartFuture;
```

Comment:
Do we really need to know if restart has never been issued? Can we initialize this future with `nullCompletedFuture`?

Reply:
In tests, we need to wait till restart gets initiated and then for it to finish. Currently, this is implemented by returning `null` to signal that no restart was initiated yet and the future when it is initiated